src/ai.ts

```diff
@@ -48,71 +235,20 @@ export async function runClaudeInference(
   }
 
   try {
-    const anthropic = apiKey ? new Anthropic({ apiKey }) : new Anthropic();
-    const response = await anthropic.messages.create({
-      messages,
-      model,
-      max_tokens: maxOutputTokens,
-      stream: true
-    });
-
-    let outputTokens = 0;
     let fullMessage = "";
-    let diffToFlush = 0;
-
-    if (streamToConsole)
-      process.stdout.write(
-        `\n\nStreaming from ${model}${
-          saveToFilepath ? ` to ${saveToFilepath}` : ""
-        }: `
-      );
-
-    for await (const chunk of response) {
-      const chunkText =
-        (chunk.type === "content_block_start" && chunk.content_block.text) ||
-        (chunk.type === "content_block_delta" && chunk.delta.text) ||
-        "";
-
-      if (chunk.type === "message_delta")
-        outputTokens += chunk.usage.output_tokens;
-
-      if (streamToConsole) process.stdout.write(chunkText);
-
-      fullMessage += chunkText;
-
-      if (saveToFilepath) {
-        diffToFlush += chunkText.length;
-
-        if (diffToFlush > NUMBER_OF_CHARACTERS_TO_FLUSH_TO_FILE) {
-          diffToFlush = 0;
-          fs.writeFileSync(saveToFilepath, (prefix || "") + fullMessage);
-        }
-      }
+    let outputTokens = 0;
+    let inputTokens = 0;
+    if (AI_PROVIDERS[provider] === undefined) {
+      throw new Error("Invalid provider");
```

Comment:
maybe this should instead be a check at the toplevel to make sure AI_MODELS_INFO has the model we're looking for?

Reply:
donesies